iOS/DuckDuckGo/AppLifecycle/AppDependencyProvider.swift

```diff
@@ -139,7 +139,22 @@ final class AppDependencyProvider: DependencyProvider {
                                                        subscriptionEndpointService: subscriptionEndpointService,
                                                        authEndpointService: authService)
 
-            let storePurchaseManager = DefaultStorePurchaseManager(subscriptionFeatureMappingCache: subscriptionFeatureMappingCache)
+            let internalUserDecider = featureFlagger.internalUserDecider
+            let subscriptionFeatureFlagger: FeatureFlaggerMapping<SubscriptionFeatureFlags> = FeatureFlaggerMapping { feature in
+                switch feature {
```

Comment:
If possible, it would be great if we could extract and unit test this. This would help prevent regressions, e.g. if someone removed the internal user check.